geopandas/plotting.py

```diff
@@ -238,52 +412,82 @@ def plot_dataframe(s, column=None, cmap=None, color=None, linewidth=1.0,
         return plot_series(s.geometry, cmap=cmap, color=color,
                            ax=ax, linewidth=linewidth, figsize=figsize,
                            **color_kwds)
+
+    if s[column].dtype is np.dtype('O'):
+        categorical = True
+
+    # Define `values` as a Series
+    if categorical:
+        if cmap is None:
+            cmap = 'Set1'
+        categories = list(set(s[column].values))
+        categories.sort()
+        valuemap = dict([(k, v) for (v, k) in enumerate(categories)])
+        values = pd.Series([valuemap[k] for k in s[column]])
     else:
-        if s[column].dtype is np.dtype('O'):
-            categorical = True
+        values = s[column]
+    if scheme is not None:
+        binning = __pysal_choro(values, scheme, k=k)
+        values = pd.Series(binning.yb)
```

Comment:
If you don't preserve the index in the newly constructed `pd.Series`, bad things happen when you index `values` with `poly_idx` if the index is not simply from [0 : sizeof(Series)-1]